README.md
## See

![Blur Glass - Google Chrome.png](https://github.com/shAdow-XJY/blur_glass/blob/master/example/assets/BlurGlass-GoogleChrome.png)

## pub

[https://pub.dev/packages/blur_glass](https://pub.dev/packages/blur_glass)

## use

```
dependencies:
    blur_glass: ^lastest_version
```

```
    BlurGlass(
        margin: EdgeInsets.all(5.0),
        padding: EdgeInsets.all(5.0),
        child: Row(
                children: [
                    Icon(Icons.star),
                    Icon(Icons.list),
                    Icon(Icons.zoom_in),
                ],
            )
    )
```

## Function creator

```
  BlurGlass({Key? key,
             required this.child,
             this.margin,
             this.padding,
             this.color,
             this.outBorderRadius = 20.0, //
             this.inBorderRadius = 30.0,  //
             this.filterX = 10.0,         //The degree of blurriness
             this.filterY = 10.0          //模糊程度
             }) : super(key: key);
```

## Link

Github:[shAdow-XJY/blur_glass: A simple blur glass widget easy to use in Flutter. (github.com)](https://github.com/shAdow-XJY/blur_glass)

Gitee:[shAdow-XJY/blur_glass (gitee.com)](https://gitee.com/shAdowPlusing/blur_glass)

CHANGELOG.md
## 0.0.1

* TODO: the first create a blur glass component.

## 0.0.2

* TODO: change use EdgeInsetsGeometry for margin and padding.

test/blur_glass_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:blur_glass/blur_glass.dart';

void main() {

}


lib/src/blurGlass.dart
// ignore_for_file: file_names

import 'dart:ui';
import 'package:flutter/material.dart';

class BlurGlass extends StatefulWidget {
  final Widget child;
  final EdgeInsetsGeometry? margin;
  final EdgeInsetsGeometry? padding;
  final Color? color;
  final double outBorderRadius;
  final double inBorderRadius;
  final double filterX;
  final double filterY;
  const BlurGlass({Key? key,
             required this.child,
             this.margin,
             this.padding,
             this.color,
             this.outBorderRadius = 20.0,
             this.inBorderRadius = 30.0,
             this.filterX = 10.0,
             this.filterY = 10.0
             }) : super(key: key);

@override
_BlurGlassState createState() => _BlurGlassState();
}

class _BlurGlassState extends State<BlurGlass> {
  @override
  Widget build(BuildContext context) {
    return ClipRRect(
      borderRadius: BorderRadius.circular(widget.outBorderRadius),
      child: BackdropFilter(
        //背景滤镜器
        filter: ImageFilter.blur(sigmaX: widget.filterX, sigmaY: widget.filterY),
        //图片模糊过滤，横向竖向都设置5.0
        child: Container(
          margin: widget.margin,//EdgeInsets.all(widget.marginValue),
          padding: widget.padding,//EdgeInsets.all(widget.paddingValue),
          decoration: BoxDecoration(
            color: widget.color??Colors.transparent.withOpacity(0.2),
            borderRadius: BorderRadius.circular(widget.inBorderRadius),
          ),
          child: widget.child,
        ),
      ),
    );
  }

}

example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:blur_glass/blur_glass.dart';
void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Blur Glass',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key,}) : super(key: key);
  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  IconButton _IconButton(String iconPath, String linkPath) {
    return IconButton(
      icon: ImageIcon(AssetImage(iconPath)),
      color: Colors.white,
      splashColor: Colors.transparent,
      highlightColor: Colors.transparent,
      onPressed: () {
        //launchUrl(Uri.parse(linkPath));
      },
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFF2E2E48),
      body: Container(
          margin: const EdgeInsets.all(10.0), //.only(top: 10),
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(10),
            image: const DecorationImage(
              fit: BoxFit.fill,
              image: AssetImage('assets/background.jpg'),
            ),
          ),
          child: Flex(
            direction: Axis.vertical,
            children: [
              Expanded(
                flex: 9,
                child: Center(
                  child: BlurGlass(
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.center,
                      mainAxisSize: MainAxisSize.min,
                      children: [
                        Text(
                          '   WELCOME\nTO  MY  BLOG!',
                          style: TextStyle(
                              color: Colors.white,
                              fontSize: 70,
                              fontWeight: FontWeight.w900,
                              shadows: [
                                BoxShadow(
                                    blurRadius: 5,
                                    color: Colors.white.withOpacity(0.54)),
                              ]),
                        ),
                        const SizedBox(height: 20),
                        const Text(
                          ' ShadowPlusing ',
                          style: TextStyle(
                            color: Colors.white,
                            fontSize: 23,
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
              ),
              Expanded(
                flex: 1,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.end,
                  children: [
                    BlurGlass(
                        margin: EdgeInsets.all(5.0),
                        padding: EdgeInsets.all(5.0),
                        child: Row(
                          children: [
                            _IconButton('assets/github.png',
                                'https://github.com/shAdow-XJY'),
                            _IconButton('assets/gitee.png',
                                'https://gitee.com/shAdowPlusing'),
                            _IconButton('assets/bilibili.png',
                                'https://space.bilibili.com/437699902'),
                          ],
                        )
                    )
                  ],
                ),
              )
            ],
          )),
    );
  }
}
